pkg/auth/service.go

```diff
@@ -44,6 +44,11 @@ type AuthorizationResponse struct {
 	Error   error
 }
 
+type NeededPermissions struct {
+	Denied       []model.Statement
+	Unauthorized []permissions.Node
```

Comment:
These are a bit too informative IMO.  Let's just keep the action name here?  Otherwise I learn details of the policy _including denied paths that I might not have known existed!_
```suggestion
	// Denied is a list of actions the user was denied for the attempt.
	Denied       []string
	// Unauthorized is a list of actions the user did not have for the attempt.
	Unauthorized []string
```